DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/EmailDelegatesController.cs

```diff
@@ -54,11 +86,39 @@ public IActionResult Index(EmailDelegatesViewModel model)
             return View("Confirmation", selectedUsers.Count);
         }
 
+        [Route("AllEmailDelegateItems")]
+        public IActionResult AllEmailDelegateItems(IEnumerable<int> selectedIds)
+        {
+            var jobGroups = jobGroupsDataService.GetJobGroupsAlphabetical();
+            var customPrompts = centreCustomPromptHelper.GetCustomPromptsForCentre(User.GetCentreId());
+            var delegateUsers = GetDelegateUserCards();
+
+            var model = new AllEmailDelegateItemsViewModel(delegateUsers, jobGroups, customPrompts, selectedIds);
+
+            return View(model);
+        }
+
         private IEnumerable<DelegateUserCard> GetDelegateUserCards()
         {
             var centreId = User.GetCentreId();
             return userService.GetDelegateUserCardsForWelcomeEmail(centreId)
                 .OrderByDescending(card => card.DateRegistered);
         }
+
+        private string? GetNewFilterBy(string? filterBy, string? filterValue)
+        {
+            if (filterBy == null && filterValue == null)
+            {
+                filterBy = Request.Cookies.ContainsKey(EmailDelegateFilterCookieName)
+                    ? Request.Cookies[EmailDelegateFilterCookieName]
+                    : null;
+            }
+            else if (filterBy?.ToUpper() == FilteringHelper.ClearString)
+            {
+                filterBy = null;
+            }
+
+            return FilteringHelper.AddNewFilterToFilterBy(filterBy, filterValue);
+        }

```

Comment:
I've been thinking about whether we should move this repeated logic from all the controllers to its own method. Is it possible to move this to a static method that all the controllers would have access to? Probably on FilteringHelper